rfc/0033-quilt-mappings-and-clean-room.md

```diff
@@ -0,0 +1,97 @@
+
+As Quilt's Yarn fork will be significantly different from upstream Yarn, this RFC also proposes changing its name to Quilt Mappings (QM).
+
+Any discussions related to migrating existing Quilt projects to other mappings standards are out of scope of this RFC, and there is - as of this writing - no intention to do so.
+
+## Motivation
+
+In order to understand the motivation for this RFC, it's important to look at how we got here. A lot has changed since Yarn was concieved in 2016, and the approach it takes isn't really relevant in today's modding landscape.
+
+### Background
+
+Yarn was originally created on August 12, 2016. It was created for a project called Open Mod Loader, the first iteration of what would be known as Fabric.
+
+It would remain in relative obscurity until October of 2018, when Yarn began tracking early snapshots for Minecraft 1.14.
+
+### Rift
+
+During the 1.13 development cycle, Forge chose to rewrite most of their scaffolding, and a stable Forge build didn't appear until Minecraft 1.14. Rift was created in response to this, as a lightweight mod loader that could do basic coremodding on Minecraft 1.13.
+
+However, Rift made use of parts of the Forge toolchain without permission. Forge declined to grant this permission to them after the fact, and Rift-related discussions were effectively banned in most modding-related communities.
+
+### Yarn's Motivation
+
+Due to the strict licensing of some parts of the Forge toolchain at the time, Fabric needed to avoid making use of Forge projects such as MCP, and thus avoid making the same mistakes that Rift did. In order to achieve this, Yarn was licensed under the [Creative Commons Zero License](https://creativecommons.org/share-your-work/public-domain/cc0/) (CC0), a public domain dedication.
+
+Making use of the CC0 in this way meant that all names in Yarn needed to be original contributions - it would have been unacceptable for a name to make it in from another mappings project, unless that project was also licensed under the CC0. For this reason, Yarn needed to take a cleanroom approach, rejecting contributions that added names from other mappings projects, and banning discussions of names from non-CC0 projects in related community spaces.
+
+### Issues With the Cleanroom
+
+While discussing how Quilt's approach to mappings should work, the Quilt Mappings Team has made the following observations:
+
+* Almost everyone that ended up taking up Fabric and Yarn were switching from other projects, generally Spigot or Forge
+* The Yarn project has never enforced a CLA, which means that there's no real certification from contributors that the names they've contributed are original
+* Some Yarn names are clearly inspired by or lifted from MCP, and couldn't have been named how they are without it
```

Comment:
Another claim without evidence. I certainly agree that in the initial founding of the project a few of these slipped in, but to my knowledge, all of those have been corrected since. I would also like to say that simply knowledge of prior names does not preclude someone from being able to come up with original names. If this statement is to remain here, surely evidence that this is true is needed.

Reply:
```suggestion
```